modules/tile-converter/src/i3s-converter/helpers/node-pages.ts

```diff
@@ -194,13 +194,15 @@ export default class NodePages {
    * @return {promise}
    */
   async save(layers0Path: string, fileMap: Object, slpk: boolean = false): Promise<void> {
-    const promises: Promise<any>[] = [];
     if (slpk) {
       for (const [index, nodePage] of this.nodePages.entries()) {
         const nodePageStr = JSON.stringify(nodePage);
         const slpkPath = join(layers0Path, 'nodepages');
-        promises.push(this.writeFile(slpkPath, nodePageStr, `${index.toString()}.json`));
-        fileMap[`nodePages/${index.toString()}.json.gz`] = `${slpkPath}.json.gz`;
+        fileMap[`nodePages/${index.toString()}.json.gz`] = await this.writeFile(
```

Comment:
It is not serializing. It is a map for the archiver, where key - path in slpk, value - path in the file system.
Parallel writing is not good for HDD/SSD hardware as I understand.

Reply:
It's fine, I have no concerns with the change. These writes are likely fast and the difference in perf is minimal. Engineering time is better spent on the real bottlenecks.

That said:

> It is not serializing.

Well, adding an `await` after every write is effectively serializing it, because the loop will no longer continue to writing the next file until the write completes.

> Parallel writing is not good for HDD/SSD hardware as I understand.

Managing parallelism to suit the hardware seems like something the operating system driver would take care of. Is it more of a general concern, or have we seen any actual information that we should change the way we write Node programs because of this?